refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -303,53 +300,53 @@ private fun SolverState.checkCallArguments(
  * [SolverState.prover] constraints.
  */
 private fun SolverState.checkConstantExpression(
-  associatedVarName: String,
-  expression: KtConstantExpression
-): ContSeq<Unit> = sideEffect {
-  solver.formulae {
-    val mayBoolean = expression.text.toBooleanStrictOrNull()
-    val mayInteger = expression.text.toBigIntegerOrNull()
-    val mayRational = expression.text.toBigDecimalOrNull()
-    when {
-      mayBoolean == true ->
-        solver.makeBooleanObjectVariable(associatedVarName)
-      mayBoolean == false ->
-        solver.booleans { not(solver.makeBooleanObjectVariable(associatedVarName)) }
-      mayInteger != null ->
-        solver.ints {
-          equal(
-            solver.makeIntegerObjectVariable(associatedVarName),
-            makeNumber(mayInteger)
-          )
+    associatedVarName: String,
+    expression: KtConstantExpression
+): ContSeq<Unit> = cont {
+    solver.formulae {
```

Comment:
What does this line do? I see we get a new `FormulaManager` here but we never seem to manipulate it. Does it create a new scope of some kind?

Reply:
It's just syntax for https://sosy-lab.github.io/java-smt/api/org/sosy_lab/java_smt/api/FormulaManager.html but it may not be in use in this scope.

Reply:
Yes, it's not being used in this scope.

Reply:
I was wondering if we could enhance the DSL to make these snippets a bit nicer with less nesting.

I.e. now we do `solver.boolean { not(booleanFormula) }` I was wondering if we couldn't add a `fun BooleanFormula.not(): BooleanFormula` inside `SolverState` since that seems to be the receiver everywhere.

That way we could eminate all the nesting everywhere, but if the nested is needed to create scopes than that's a different story.

Reply:
The problem is that you need either a `FormulaManager` or one of the `BooleanFormulaManager`, `NumeralFormulaManager`... to create new formulae. So it's not possible simply do `BooleanFormula.not()`, which would be the nicest DSL.

Reply:
You can hide that way in `SolverState`, no? This is what I had in mind.

```
class SolverState(...) {
   fun BooleanFormula.not(): BooleanFormula =
     booleans { not(this@not) }
}
```

Since it's defined inside `SolverState` it has access to `booleans { }`, right?

And since all the functions defined in `ConstraintChecker.kt` are defined as extensions on `SolverState` you would have that syntax available inside the method bodies.
It's similar to what soon would be solve-able with multiple context receivers.

It's just a suggestion because the nesting seems very annoying if it's not needed for wrapping code in scopes.